core/modules/qdisp/Executive.h

```diff
@@ -44,16 +46,39 @@ class XrdSsiService;
 namespace lsst {
 namespace qserv {
 namespace qdisp {
+
+class JobQuery;
 class MessageStore;
 class QueryResource;
-class ResponseRequester;
+
+/** Description of a job managed by the executive
+ */
+class JobDescription {
+public:
+    JobDescription(int id, ResourceUnit resource, std::string payload,
+        std::shared_ptr<ResponseHandler> respHandler)
```

Comment:
`resource`, `payload` and `respHandler` can be passed as const ref for efficiency